testsuite/MDAnalysisTests/coordinates/test_lammps.py

```diff
@@ -189,177 +183,169 @@ def test_Writer_atoms(self, attr, LAMMPSDATA_partial):
                             iframe * self.dt,
                             err_msg="Time for frame {0} (dt={1}) is wrong.".format(
-                iframe, self.dt))
+                                iframe, self.dt))
 
-    def test_LAMMPSDCDReader_set_dt(self, dt=1500.):
+    def test_LAMMPSDCDReader_set_dt(self, u, dt=1500.):
         u = mda.Universe(self.topology, self.trajectory, format=self.format,
                          dt=dt)
-        iframe = self.get_frame_from_end(1)
-        assert_almost_equal(u.trajectory[iframe].time, iframe*dt,
+        iframe = self.get_frame_from_end(1, u)
+        assert_almost_equal(u.trajectory[iframe].time, iframe * dt,
                             err_msg="setting time step dt={0} failed: "
-                            "actually used dt={1}".format(
-                dt, u.trajectory._ts_kwargs['dt']))
+                                    "actually used dt={1}".format(
+                                dt, u.trajectory._ts_kwargs['dt']))
 
     def test_wrong_time_unit(self):
         def wrong_load(unit="nm"):
-            return mda.Universe(self.topology, self.trajectory, format=self.format,
+            return mda.Universe(self.topology, self.trajectory,
+                                format=self.format,
                                 timeunit=unit)
+
         with pytest.raises(TypeError):
             wrong_load()
 
     def test_wrong_unit(self):
         def wrong_load(unit="GARBAGE"):
-            return mda.Universe(self.topology, self.trajectory, format=self.format,
+            return mda.Universe(self.topology, self.trajectory,
+                                format=self.format,
                                 timeunit=unit)
+
         with pytest.raises(ValueError):
             wrong_load()
 
 
-class TestLAMMPSDCDWriter(TestCase, RefLAMMPSDataDCD):
+class TestLAMMPSDCDWriter(RefLAMMPSDataDCD):
     flavor = 'LAMMPS'
 
-    def setUp(self):
-        self.u = mda.Universe(self.topology, self.trajectory,
-                              format=self.format)
-        # dummy output file
-        ext = os.path.splitext(self.trajectory)[1]
-        self.tmpdir = tempdir.TempDir()
-        self.outfile = os.path.join(self.tmpdir.name,  'lammps-writer-test' + ext)
-
-    def tearDown(self):
-        try:
-            os.unlink(self.outfile)
-        except OSError:
-            pass
-        del self.u
-        del self.tmpdir
-
-    def test_Writer_is_LAMMPS(self):
-        with mda.Writer(self.outfile, n_atoms=self.u.atoms.n_atoms,
-                       format=self.format) as W:
-            assert_(W.flavor, self.flavor)
-
-    def test_Writer(self, n_frames=3):
-        W = mda.Writer(self.outfile, n_atoms=self.u.atoms.n_atoms,
+    @pytest.fixture(scope='class')
+    def u(self):
+        return mda.Universe(self.topology, self.trajectory, format=self.format)
+
+    def test_Writer_is_LAMMPS(self, u, tmpdir):
+        outfile = str(tmpdir) + 'lammps-writer-test' + \
```

Comment:
`outfile = os.path.join(str(tmpdir), 'lammps-writer-test' + ext)`

Don't join paths manually!

Reply:
In general those manual line breaks with `\` are also not that nice and should be avoided if possible.